File: 2761-prime-pairs-with-target-sum/2761-prime-pairs-with-target-sum.py
class Solution:
    def findPrimePairs(self, n: int) -> List[List[int]]:
        # sieve of Eratosthenes
        vis = [False] * (n + 1)
        for i in range(2, int(n**0.5) + 1):
            if not vis[i]:
                for j in range(2 * i, n + 1, i):
                    vis[j] = True

        ans = []

        for i in range(2, n // 2 + 1):
            x = i
            y = n - i

            if not vis[x] and not vis[y] and x <= y:
                ans.append([x, y])

        return ans

File: 2761-prime-pairs-with-target-sum/2761-prime-pairs-with-target-sum.cpp
class Solution {
public:
  vector<vector<int>> findPrimePairs(int n) {
        // steive of eratharos
        vector<bool> vis(n + 1, true);
        for (int i = 2; i * i <= n; i++){
            if (vis[i]){
                for (int j = 2 * i; j <= n; j += i){
                    vis[j] = false;
                }
            }        
        }
        
        vector<vector<int>> ans;
        
        for (int i = 2; i <= n / 2; i++){
            int x = i;
            int y = n - i;
            
            if (vis[x] && vis[y] && x <= y){
                ans.push_back({x, y});
            }
        }
        
        return ans;
    }
};